Doppler.PushContact/Controllers/PushContactController.cs

```diff
@@ -132,5 +133,12 @@ public async Task<IActionResult> GetMessageDetails([FromRoute] string domain, [F
                 messageDetails.NotDelivered
             });
         }
+
+        [HttpGet]
+        [Route("push-contacts/messages")]
+        public async Task<IEnumerable<MessageApiModel>> GetMessages([FromQuery] DateTimeOffset from, [FromQuery] DateTimeOffset to)
```

Comment:
Maybe you can define a model for the page:

```suggestion
        public async Task<MessagePageApiModel> GetMessages([FromQuery] DateTimeOffset from, [FromQuery] DateTimeOffset to)
```

Or 

```suggestion
        public async Task<ApiPage<MessageApiModel>> GetMessages([FromQuery] DateTimeOffset from, [FromQuery] DateTimeOffset to)
```

In this way, you can include some extra information (ie. continuation token) in the result object.

In general, returning an array is not a good idea because adding more properties is not possible, try to always wrap the collection with an object.

Reply:
it's an awesome idea!